web_ui/prometheus.go

```diff
@@ -84,6 +84,28 @@ func init() {
 	prometheus.MustRegister(version.NewCollector(strings.ReplaceAll(appName, "-", "_")))
 }
 
+const (
+	MaxLabelLimit            = 10
+	MaxLabelNameLengthLimit  = 256
+	MaxLabelValueLengthLimit = 4096
+	MaxSampleLimit           = 800
+)
+
+func setCardinalityLimits(cfg *config.ScrapeConfig) {
+	if param.Monitoring_LabelLimit.GetInt() > 0 && param.Monitoring_LabelLimit.GetInt() < MaxLabelLimit {
```

Comment:
Instead of failing to set values when user provided configuration is bad, we should
a) fall back to the default and use that to configure the limits
b) log a warning for the user that their provided value of %d is bad and that we're falling back to use the default of %d.